File: hr.py
#!/usr/bin/python3
'''
 hr - horizontal bar for your terminal
 (c) Copyright 2019 Bartosz Mierzynski
'''

from sys import argv, exit, stdout
from os import get_terminal_size

def main():
	try:
		cols = get_terminal_size().columns
	except OSError:
		cols = 80
	if len(argv) < 2:
		while cols:
			stdout.write('#');
			cols -= 1
		stdout.write('\n');
	else:
		for j in range(1, len(argv)):
			length = len(argv[j])
			for i in range(0, cols):
				stdout.write(argv[j][i%length])
			stdout.write('\n');
	return 0

if __name__ == "__main__":
	exit(main())

File: hr.lua
#!/usr/bin/env lua
--[[ 
 hr - writes horizontal bar on terminal
 (c) Copyright 2019 Bartosz Mierzynski
]]

local function main()
	cols = tonumber(os.getenv("COLUMNS"));
	if(cols == nil)
	then
		cols = 80
	end
	if(#arg == 0)
	then
		while(cols > 0)
		do
			io.write('#')
			cols = cols - 1
		end
		io.write('\n')
	else
		for j = 1,#arg, 1
		do
			for i = 0, cols-1, 1
			do
				io.write(arg[j]:sub(i%#arg[j]+1,i%#arg[j]+1))
			end
			io.write('\n')
		end
	end
	return 0
end

os.exit(main())

